packages/@tinacms/schema-tools/src/schema/addNamespaceToSchema.test.ts

```diff
@@ -0,0 +1,352 @@
+    const node = {
+      name: 'root',
+      mixedArray: [
+        { name: 'child1' },
+        'stringElement',
+        { otherProp: 'childData' },
+      ],
+    }
+    const result = addNamespaceToSchema(node)
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      mixedArray: [
+        { name: 'child1', namespace: ['child1'] },
+        'stringElement',
+        { otherProp: 'childData' },
+      ],
+    })
+  })
+
+  test('should maintain original properties not related to name or namespace', () => {
+    const node = {
+      name: 'node1',
+      otherProp: 'value',
+    }
+    const result = addNamespaceToSchema(node)
+    expect(result).toEqual({
+      name: 'node1',
+      namespace: [],
+      otherProp: 'value',
+    })
+  })
+
+  test('should correctly generate separate namespaces for nested objects', () => {
+    const node = {
+      name: 'root',
+      children: [
+        { name: 'child1' },
+        { name: 'child2', children: [{ name: 'grandchild1' }] },
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        { name: 'child1', namespace: ['child1'] },
+        {
+          name: 'child2',
+          namespace: ['child2'],
+          children: [
+            { name: 'grandchild1', namespace: ['child2', 'grandchild1'] },
+          ],
+        },
+      ],
+    })
+
+    // Check if namespaces are not affecting each other
+    expect((result.children[0] as Node).namespace).toEqual(['child1'])
+    expect((result.children[1] as Node).namespace).toEqual(['child2'])
+    expect((result.children?.[1]?.children?.[0] as Node).namespace).toEqual([
+      'child2',
+      'grandchild1',
+    ])
+  })
+
+  test('should generate separate namespaces for sibling nodes', () => {
+    const node = {
+      name: 'root',
+      children: [
+        { name: 'child1' },
+        { name: 'child2', children: [{ name: 'child1' }] }, // Repeated 'child1' name
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that namespaces are correctly separated
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        { name: 'child1', namespace: ['child1'] },
+        {
+          name: 'child2',
+          namespace: ['child2'],
+          children: [
+            { name: 'child1', namespace: ['child2', 'child1'] }, // Should not be affected by the first 'child1'
+          ],
+        },
+      ],
+    })
+
+    // Ensure namespaces do not share references
+    expect((result.children[0] as Node).namespace).toEqual(['child1'])
+    expect((result.children[1] as Node).namespace).toEqual(['child2'])
+    expect((result.children?.[1]?.children?.[0] as Node).namespace).toEqual([
+      'child2',
+      'child1',
+    ])
+  })
+
+  test('should generate unique namespaces for nested nodes with similar names', () => {
```

Comment:
Should namespaces be unique? This test does not test for that.

Reply:
👍🏻 updated with more accurate name